tls/s2n_signature_algorithms.c

```diff
@@ -22,10 +22,13 @@
 
 static int s2n_sig_hash_algs_pairs_set(struct s2n_sig_hash_alg_pairs *sig_hash_algs, uint8_t sig_alg, uint8_t hash_alg)
 {
-    S2N_ERROR_IF(hash_alg >= TLS_HASH_ALGORITHM_COUNT, S2N_ERR_HASH_INVALID_ALGORITHM);
-    S2N_ERROR_IF(sig_alg >= TLS_SIGNATURE_ALGORITHM_COUNT, S2N_ERR_INVALID_SIGNATURE_ALGORITHM);
-    
-    sig_hash_algs->matrix[sig_alg][hash_alg] = 1;
+    /* The RFC requires these values to be between 0 and 255 */
+    S2N_ERROR_IF(hash_alg > 255, S2N_ERR_HASH_INVALID_ALGORITHM);
+    S2N_ERROR_IF(sig_alg > 255, S2N_ERR_INVALID_SIGNATURE_ALGORITHM);
+
+    if (hash_alg < TLS_HASH_ALGORITHM_COUNT && sig_alg < TLS_SIGNATURE_ALGORITHM_COUNT) {
```

Comment:
I think a comment here explaining that we're skipping values s2n does not recognize would be useful. A reader in the future may not catch that immediately.